bundler/spec/install/gemfile_spec.rb

```diff
@@ -27,6 +27,35 @@
       ENV["BUNDLE_GEMFILE"] = "NotGemfile"
       expect(the_bundle).to include_gems "myrack 1.0.0"
     end
+
+    it "respects lock_file and BUNDLE_GEMFILE_LOCK" do
+      gemfile bundled_app("NotGemfile"), <<-G
+        lock_file "ReallyNotGemfile.lock"
+        source "https://gem.repo1"
+        gem 'myrack'
+      G
+
+      bundle :install, gemfile: bundled_app("NotGemfile")
+
+      ENV["BUNDLE_GEMFILE"] = "NotGemfile"
+      ENV["BUNDLE_GEMFILE_LOCK"] = "ReallyNotGemfile.lock"
+      expect(the_bundle).to include_gems "myrack 1.0.0"
+    end
+
+    it "respects BUNDLE_GEMFILE_LOCK during bundle install" do
+      ENV["BUNDLE_GEMFILE_LOCK"] = "ReallyNotGemfile.lock"
+
+      gemfile bundled_app("NotGemfile"), <<-G
+        source "https://gem.repo1"
+        gem 'myrack'
+      G
+
+      bundle :install, gemfile: bundled_app("NotGemfile")
+      expect(File.exist?(bundled_app("ReallyNotGemfile.lock"))).to eq true
```

Comment:
Can we use this?

```suggestion
      expect(bundled_app("ReallyNotGemfile.lock")).to exist
```

Reply:
That looks better to me. I merged it into the PR.